vcd/resource_vcd_edgegateway_settings.go

```diff
@@ -0,0 +1,228 @@
+package vcd
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+)
+
+func resourceVcdEdgeGatewaySettings() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVcdEdgeGatewaySettingsCreate,
+		Read:   resourceVcdEdgeGatewaySettingsRead,
+		Update: resourceVcdEdgeGatewaySettingsUpdate,
+		Delete: resourceVcdEdgeGatewaySettingsDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceVcdEdgeGatewaySettingsImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"org": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Description: "The name of organization to use, optional if defined at provider " +
+					"level. Useful when connected as sysadmin working across different organizations",
+			},
+			"vdc": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The name of VDC to use, optional if defined at provider level",
+			},
+			"edge_gateway_name": &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				Description:  "Name of the edge gateway. Required when 'edge_gateway_id' is not set",
+				ExactlyOneOf: []string{"edge_gateway_id", "edge_gateway_name"},
+			},
+			"edge_gateway_id": &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				Description:  "ID of the edge gateway. Required when 'edge_gateway_name' is not set",
+				ExactlyOneOf: []string{"edge_gateway_id", "edge_gateway_name"},
+			},
+			"lb_enabled": &schema.Schema{
+				Type:        schema.TypeBool,
+				Default:     false,
+				Optional:    true,
+				Description: "Enable load balancing. (Disabled by default)",
+			},
+			"lb_acceleration_enabled": &schema.Schema{
+				Type:        schema.TypeBool,
+				Default:     false,
+				Optional:    true,
+				Description: "Enable load balancer acceleration. (Disabled by default)",
+			},
+			"lb_logging_enabled": &schema.Schema{
+				Type:        schema.TypeBool,
+				Default:     false,
+				Optional:    true,
+				Description: "Enable load balancer logging. (Disabled by default)",
+				// Due to a vCD bug, this field can only be changed by a system administrator
```

Comment:
Let's use the new abbreviation capitalization `VCD` (VMware Cloud Director) in all new messages.
```suggestion
				// Due to a VCD bug, this field can only be changed by a system administrator
```

Reply:
fixed